api/v1alpha1/module_types.go

```diff
@@ -92,42 +93,112 @@ type KernelMapping struct {
-	// in addition to the default ones.
-	AdditionalVolumes []v1.Volume `json:"additionalVolumes"`
+	RawArgs *ModprobeArgs `json:"rawArgs,omitempty"`
+}
 
+type DriverContainerContainerSpec struct {
+	// Build contains build instructions.
 	// +optional
-	// DevicePlugin allows overriding some properties of the container that deploys the device plugin on the node.
-	// Name is ignored and is set automatically by the OOT Operator.
-	DevicePlugin *v1.Container `json:"devicePlugin"`
+	Build *Build `json:"build,omitempty"`
 
-	// DriverContainer allows overriding some properties of the container that deploys the driver on the node.
-	// Name and image are ignored and are set automatically by the OOT Operator.
-	DriverContainer v1.Container `json:"driverContainer"`
+	// ContainerImage is a top-level field
+	// +optional
+	ContainerImage string `json:"containerImage,omitempty"`
+
+	// Image pull policy.
+	// One of Always, Never, IfNotPresent.
+	// Defaults to Always if :latest tag is specified, or IfNotPresent otherwise.
+	// Cannot be updated.
+	// More info: https://kubernetes.io/docs/concepts/containers/images#updating-images
+	// +optional
+	ImagePullPolicy v1.PullPolicy `json:"imagePullPolicy,omitempty" protobuf:"bytes,14,opt,name=imagePullPolicy,casttype=PullPolicy"`
 
 	// KernelMappings is a list of kernel mappings.
 	// When a node's labels match Selector, then the OOT Operator will look for the first mapping that matches its
 	// kernel version, and use the corresponding container image to run the DriverContainer.
+	// +kubebuilder:validation:MinItems=1
 	KernelMappings []KernelMapping `json:"kernelMappings"`
 
-	// Selector describes on which nodes the Module should be loaded.
-	Selector map[string]string `json:"selector"`
+	// Modprobe is a set of properties to customize which module modprobe loads and with which properties.
+	Modprobe ModprobeSpec `json:"modprobe"`
+}
+
+type DriverContainerSpec struct {
+	// Container holds the properties for the driver container that runs modprobe.
+	Container DriverContainerContainerSpec `json:"container"`
+
+	// +optional
+	ImageRepoSecret *v1.LocalObjectReference `json:"imageRepoSecret,omitempty"`
+
+	// +optional
+	// ServiceAccountName is the name of the ServiceAccount to use to run this pod.
+	// More info: https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/
+	ServiceAccountName string `json:"serviceAccountName,omitempty"`
+}
+
+type DevicePluginContainerSpec struct {
+	// Image is the name of the container image that the device plugin container will run.
+	Image string `json:"image"`
+
+	// VolumeMounts is a list of volume mounts that are appended to the default ones.
+	// +optional
+	VolumeMounts []v1.VolumeMount `json:"volumeMounts,omitempty"`
+}
+
+type DevicePluginSpec struct {
+	Container DevicePluginContainerSpec `json:"container"`
+
+	ImagePullSecrets []v1.LocalObjectReference `json:"imagePullSecrets,omitempty"`
```

Comment:
Why is ImagePullSecret specific to the DevicePlugin and not the whole Module?

Reply:
I agree that it will be clearer if we only have `ImageRepoSecret` at the module level to contain all the auth for all image registries in a single dockerconfig.json file.

Isn't what we have agreed on?

Reply:
Switched to this.